pkg/controller/machinepool/awsactuator.go

```diff
@@ -531,3 +728,86 @@ func (a *AWSActuator) validateSubnets(subnets map[string]ec2.Subnet, pool *hivev
 	}
 	return subnetsByAvailabilityZone, nil
 }
+
+func awsMachineTemplate(infraName, ami string, nodePool *hivev1.MachinePool, subnetRef *capiaws.AWSResourceReference, controlPlaneNamespace, zone string) *capiaws.AWSMachineTemplate {
+	securityGroups := []capiaws.AWSResourceReference{{
+		Filters: []capiaws.Filter{{
+			Name:   "tag:Name",
+			Values: []string{fmt.Sprintf("%s-worker-sg", infraName)},
+		}},
+	}}
+
+	instanceProfile := fmt.Sprintf("%s-worker-profile", infraName)
+
+	instanceType := nodePool.Spec.Platform.AWS.InstanceType
+
+	awsMachineTemplate := &capiaws.AWSMachineTemplate{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				machinePoolNameLabel: nodePool.Name,
+			},
+			Namespace: controlPlaneNamespace,
+		},
+		Spec: capiaws.AWSMachineTemplateSpec{
+			Template: capiaws.AWSMachineTemplateResource{
+				Spec: capiaws.AWSMachineSpec{
+					UncompressedUserData: pointer.BoolPtr(true),
+					CloudInit: capiaws.CloudInit{
+						InsecureSkipSecretsManager: true,
+						SecureSecretsBackend:       "secrets-manager",
+					},
+					IAMInstanceProfile: instanceProfile,
+					InstanceType:       instanceType,
+					AMI: capiaws.AWSResourceReference{
+						ID: pointer.StringPtr(ami),
+					},
+					AdditionalSecurityGroups: securityGroups,
+					Subnet:                   subnetRef,
+				},
+			},
+		},
+	}
+	awsMachineTemplate.SetName(fmt.Sprintf("%s-worker-%s", infraName, zone))
+	return awsMachineTemplate
+}
+
+func (a *AWSActuator) awsMachinePool(pool *hivev1.MachinePool) (*types.MachinePool, error) {
+	computePool := baseMachinePool(pool)
+	computePool.Platform.AWS = &installertypesaws.MachinePool{
+		AMIID:        a.amiID,
+		InstanceType: pool.Spec.Platform.AWS.InstanceType,
+		EC2RootVolume: installertypesaws.EC2RootVolume{
+			IOPS:      pool.Spec.Platform.AWS.EC2RootVolume.IOPS,
+			Size:      pool.Spec.Platform.AWS.EC2RootVolume.Size,
+			Type:      pool.Spec.Platform.AWS.EC2RootVolume.Type,
+			KMSKeyARN: pool.Spec.Platform.AWS.EC2RootVolume.KMSKeyARN,
```

Comment:
Totally not something to worry about now, but this made me wonder why MachinePool.Spec.Platform.AWS.EC2RootVolume isn't already of type installertypesaws.EC2RootVolume so we can just do a straight assignment here...